rust/sdk-processor/src/utils/database.rs

```diff
@@ -0,0 +1,329 @@
+    }
+
+    Ok(())
+}
+
+pub async fn execute_with_better_error<U>(
+    pool: ArcDbPool,
+    query: U,
+    mut additional_where_clause: Option<&'static str>,
+) -> QueryResult<usize>
+where
+    U: QueryFragment<Backend> + diesel::query_builder::QueryId + Send,
+{
+    let original_query = diesel::debug_query::<Backend, _>(&query).to_string();
+    // This is needed because if we don't insert any row, then diesel makes a call like this
+    // SELECT 1 FROM TABLE WHERE 1=0
+    if original_query.to_lowercase().contains("where") {
+        additional_where_clause = None;
+    }
+    let final_query = UpsertFilterLatestTransactionQuery {
+        query,
+        where_clause: additional_where_clause,
+    };
+    let debug_string = diesel::debug_query::<Backend, _>(&final_query).to_string();
+    tracing::debug!("Executing query: {:?}", debug_string);
+    let conn = &mut pool.get().await.map_err(|e| {
+        tracing::warn!("Error getting connection from pool: {:?}", e);
+        diesel::result::Error::DatabaseError(
+            diesel::result::DatabaseErrorKind::UnableToSendCommand,
+            Box::new(e.to_string()),
+        )
+    })?;
+    let res = final_query.execute(conn).await;
+    if let Err(ref e) = res {
+        tracing::warn!("Error running query: {:?}\n{:?}", e, debug_string);
+    }
+    res
+}
+
+/// Returns the entry for the config hashmap, or the default field count for the insert
+/// Given diesel has a limit of how many parameters can be inserted in a single operation (u16::MAX),
+/// we default to chunk an array of items based on how many columns are in the table.
+pub fn get_config_table_chunk_size<T: field_count::FieldCount>(
+    table_name: &str,
+    per_table_chunk_sizes: &AHashMap<String, usize>,
+) -> usize {
+    per_table_chunk_sizes
+        .get(table_name)
+        .copied()
+        .unwrap_or_else(|| MAX_DIESEL_PARAM_SIZE / T::field_count())
+}
+
+pub async fn execute_with_better_error_conn<U>(
+    conn: &mut MyDbConnection,
+    query: U,
+    mut additional_where_clause: Option<&'static str>,
+) -> QueryResult<usize>
+where
+    U: QueryFragment<Backend> + diesel::query_builder::QueryId + Send,
+{
+    let original_query = diesel::debug_query::<Backend, _>(&query).to_string();
+    // This is needed because if we don't insert any row, then diesel makes a call like this
+    // SELECT 1 FROM TABLE WHERE 1=0
+    if original_query.to_lowercase().contains("where") {
+        additional_where_clause = None;
+    }
+    let final_query = UpsertFilterLatestTransactionQuery {
+        query,
+        where_clause: additional_where_clause,
+    };
+    let debug_string = diesel::debug_query::<Backend, _>(&final_query).to_string();
+    tracing::debug!("Executing query: {:?}", debug_string);
+    let res = final_query.execute(conn).await;
+    if let Err(ref e) = res {
+        tracing::warn!("Error running query: {:?}\n{:?}", e, debug_string);
+    }
+    res
+}
+
+async fn execute_or_retry_cleaned<U, T>(
```

Comment:
we should rethink about the retry strategy here.

Reply:
Added as a task